lib/reply.js

```diff
@@ -186,12 +187,8 @@ Reply.prototype.send = function (payload) {
 
 Reply.prototype.getHeader = function (key) {
   key = key.toLowerCase()
-  const res = this.raw
-  let value = this[kReplyHeaders][key]
-  if (value === undefined && res.hasHeader(key)) {
-    value = res.getHeader(key)
-  }
-  return value
+
+  return this[kReplyHeaders][key] ?? this.raw.getHeader(key)
```

Comment:
It's not clear to me if this change is equivalent. `??` will return the righthand side when the left is `undefined` _or_ `null`. The original check is specifically for `undefined`.

Reply:
In node's types `null` is invalid value for a header `OutgoingMessage.setHeader(name: string, value: number | string | ReadonlyArray<string>): this;`, but node docs says it can be `<any>`. If in fastify `null` is valid value for a header I can change this line to:

```js
return this[kReplyHeaders][key] !== undefined ? this[kReplyHeaders][key] : this.raw.getHeader(key)
```

Reply:
In my opinion, the original code is much easier to read.

Reply:
Just to make sure we are talking about the same code:
This code is more readable:
```js
  const res = this.raw
  let value = this[kReplyHeaders][key]
  if (value === undefined && res.hasHeader(key)) {
    value = res.getHeader(key)
  }
  return value
```

than this code?

```js
return this[kReplyHeaders][key] !== undefined ? this[kReplyHeaders][key] : this.raw.getHeader(key)
```

Reply:
That is correct. There's a lot to keep track of in the single line version. The multiple line version literally shows what work is being done to determine the result `value`.

Reply:
reverted